javascript/aws-lambda/security/mysql-sqli.yaml

```diff
@@ -0,0 +1,62 @@
+rules:
+- id: mysql-sqli
+  severity: WARNING
+  message: >-
+    Detected SQL statement that is tainted by `$EVENT` object. This could lead to SQL injection if the variable is user-controlled
+    and not properly sanitized. In order to prevent SQL injection,
+    used parameterized queries or prepared statements instead.
+    You can use parameterized statements like so:
+    `connection.query('SELECT $1 from table', [userinput])`
+  languages:
+  - javascript
+  - typescript
+  mode: taint
+  metadata:
+    references:
+      - https://www.npmjs.com/package/mysql2
+    category: security
+    technology:
+      - aws-lambda
+      - mysql
+      - mysql2
+  pattern-sources:
+  - patterns:
+    - pattern-either:
+      - pattern-inside: |
+          exports.handler = function ($EVENT, ...) {
+            ...
+          }
+      - pattern-inside: |
+          function $FUNC ($EVENT, ...) {...}
+          ...
+          exports.handler = $FUNC
+      - pattern-inside: |
+          $FUNC = function ($EVENT, ...) {...}
+          ...
+          exports.handler = $FUNC
+    - pattern: $EVENT
+  pattern-sinks:
+  - patterns:
+    - pattern: $QUERY
+    - pattern-either:
+      - pattern-inside: $POOL.query($QUERY, ...)
+      - pattern-inside: $POOL.execute($QUERY, ...)
+    - pattern-either:
+      - pattern-inside: |
+          require('mysql')
+          ...
+      - pattern-inside: |
+          require('mysql2')
+          ...
+      - pattern-inside: |
+          require('mysql2/promise')
+          ...
+      - pattern-inside: |
```

Comment:
Is this leftover from the previous rule?

Reply:
yes! thank you :+1: